### ISSUE TITLE
OracleDatabase doesn't produce correct CYCLE sequence flag information in code generation

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12640]
When using these two sequences in Oracle:

```sql
CREATE SEQUENCE s_12638_y CACHE 50 START WITH 0 INCREMENT BY 2 CYCLE MINVALUE 0 MAXVALUE 100;
CREATE SEQUENCE s_12638_n NOCACHE NOCYCLE;
```

The generated code is:

```java
/**
 * The sequence <code>TEST.S_12638_N</code>
 */
public static final Sequence<BigInteger> S_12638_N = Internal.createSequence("S_12638_N", Test.TEST, SQLDataType.DECIMAL_INTEGER(38).nullable(false), null, null, null, null, false, 0);

/**
 * The sequence <code>TEST.S_12638_Y</code>
 */
public static final Sequence<Byte> S_12638_Y = Internal.createSequence("S_12638_Y", Test.TEST, SQLDataType.TINYINT.nullable(false), null, 2, 0, 100, false, 50);
```

As can be seen, while the `S_12638_N` sequence is generated correctly, `S_12638_Y` is lacking:

- `START WITH` information
- `CYCLE` information

See also: https://github.com/jOOQ/jOOQ/issues/12638

**Comment by lukaseder** [comment]
`SYS.ALL_SEQUENCES` doesn't contain any `START_WITH` information, which is why we always produce `1`.

**Comment by lukaseder** [comment]
It's probably just not such an interesting flag, because it never reproduces after the first `NEXTVAL` call, e.g. `ALTER SEQUENCE s RESTART` doesn't restart at `START WITH`, but at `MINVALUE`

**Comment by lukaseder** [comment]
We're simply not reading the `CYCLE_FLAG` column...

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.5 (https://github.com/jOOQ/jOOQ/issues/12641), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12642)
